addons/project_account/models/project_project.py

```diff
@@ -35,13 +35,13 @@ def _get_costs_items_from_purchase(self, domain, profitability_items, with_actio
         bills_move_line_read = self._cr.dictfetchall()
         if bills_move_line_read:
             # Get conversion rate from currencies to currency of the current company
-            currency_ids = {bml['currency_id'] for bml in bills_move_line_read + [{'currency_id': self.currency_id.id}]}
-            rates = self.env['res.currency'].browse(list(currency_ids))._get_rates(self.company_id or self.env.company, fields.Date.context_today(self))
-            conversion_rates = {cid: self.currency_id.rate / rate_from for cid, rate_from in rates.items()}
+            currency_ids = {bml['currency_id'] for bml in bills_move_line_read}
```

Comment:
Using a `set` here introduces nondeterminism in the prefetching below, which might be painful when it comes to debugging. I recommend using an `OrderedSet` instead, or something equivalent.